### PR TITLE
py-mikado: Add version 1.2.4

### LINKED ISSUE #20552: Installation issue: py-mikado
I have detected two problems.
1) The version of py-biopython and py-pysam specified in depends_on is the version that spack does not support.
2) py-mikado will result in a Fetch Error.

I can handle the error 1), but not 2).

### PR DESCRIPTION
**Pull request by t-nojiri**
Fixes #20552

I specified the version number for py-pandas because the following message was output at build time. py-pandas requires python version 3.7.1:, but spec asked for 3.6.12

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-biopython/package.py [hunk 1]
```diff
@@ -15,6 +15,7 @@ class PyBiopython(PythonPackage):
     homepage = "http://biopython.org/wiki/Main_Page"
     url      = "http://biopython.org/DIST/biopython-1.65.tar.gz"
 
+    version('1.78', sha256='1ee0a0b6c2376680fea6642d5080baa419fd73df104a62d58a8baf7a8bbe4564')
     version('1.73', sha256='70c5cc27dc61c23d18bb33b6d38d70edc4b926033aea3b7434737c731c94a5e0')
     version('1.70', sha256='4a7c5298f03d1a45523f32bae1fffcff323ea9dce007fb1241af092f5ab2e45b')
     version('1.65', sha256='463cc81db84e9bfcdfb15629511c81ed556a6c0287e670dbfe80f03c65d2a88e')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-biopython/package.py:21]: The latest version requires:
```python
depends_on('python@3.6:', type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-mikado/package.py [hunk 2]
```diff
@@ -12,9 +12,9 @@ class PyMikado(PythonPackage):
        the best models in each locus."""
 
     homepage = "https://github.com/EI-CoreBioinformatics/mikado"
-    git      = "git://github.com/EI-CoreBioinformatics/mikado"
+    url = "https://pypi.io/packages/source/M/Mikado/Mikado-1.2.4.tar.gz"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-mikado/package.py]:
```suggestion
    pypi = "Mikado/Mikado-1.2.4.tar.gz"
```

File: var/spack/repos/builtin/packages/py-mikado/package.py [hunk 3]
```diff
@@ -38,7 +38,7 @@ class PyMikado(PythonPackage):
     depends_on('py-tabulate', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-mikado/package.py:37]: Check the `requirements.txt`, many of these versions have changed

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-mikado/package.py [hunk 4]
```diff
@@ -38,7 +38,7 @@ class PyMikado(PythonPackage):
...
     depends_on('py-msgpack@1.0.0:', type=('build', 'run'))
     depends_on('py-python-rapidjson@0.8:', type=('build', 'run'))
-    depends_on('py-pandas', type=('build', 'run'))
+    depends_on('py-pandas@:1.1.5', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-mikado/package.py]: I would undo this change. If you're using Python 3.8, then any version of pandas should work. If you're having concretization issues, you may want to try the new concretizer, or explicitly set things like this in your `packages.yaml`.

**Comment by t-nojiri** [comment]
I'm using Python 3.8.6.
Reverting the py-pandas fix causes conflicts.
$ spack debug report
* ** Spack: ** 0.16.0-448-e6d462589
* ** Python: ** 3.8.6
* ** Platform: ** linux-rhel8-thunderx2
* ** Concreteizer: ** original
$ spack install py-mikado %gcc@8.3.1
:
py-pandas requires python version 3.7.1:, but spec asked for 3.6.12


The new concretizer gave the following error.
$ spack debug report
* **Spack:** 0.16.0-448-e6d462589
* **Python:** 3.8.7
* **Platform:** linux-rhel8-thunderx2
* **Concretizer:** clingo
$ spack install py-mikado %gcc@8.3.1
==> Error: 'gdb'


Should I undo the py-pandas fix?

**Comment by t-nojiri** [comment]
I didn't check requirements.txt.
frozendict wasn't in the spack list, but I modified everything else and I was able to build py-mikado.

I will push this fix again.

File: var/spack/repos/builtin/packages/py-biopython/package.py (hunk 1 above)
Comment (t-nojiri) [reply var/spack/repos/builtin/packages/py-biopython/package.py:21]: I reflected the fix for py-biopython.

File: var/spack/repos/builtin/packages/py-mikado/package.py [hunk 5]
```diff
@@ -12,34 +12,29 @@ class PyMikado(PythonPackage):
...
-    depends_on('py-msgpack@1.0.0:', type=('build', 'run'))
-    depends_on('py-python-rapidjson@0.8:', type=('build', 'run'))
-    depends_on('py-pandas', type=('build', 'run'))
-    depends_on('py-jsonref', type=('build', 'run'))
-    depends_on('py-pysam@0.15.3:', type=('build', 'run'))
-    depends_on('py-toml', type=('build', 'run'))
-    depends_on('py-tomlkit', type=('build', 'run'))
+    depends_on('py-ujson', type=('build', 'run'))
+    depends_on('py-simplejson', type=('build', 'run'))
+    depends_on('python@3.6:', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-mikado/package.py]:
```suggestion
    depends_on('python@3.4:', type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-mikado/package.py [hunk 6]
```diff
@@ -12,34 +12,29 @@ class PyMikado(PythonPackage):
...
-    git      = "git://github.com/EI-CoreBioinformatics/mikado"
+    pypi = "Mikado/Mikado-1.2.4.tar.gz"
 
-    version('20201016', commit='2501ba995d85e62e47165fe72592666a4c79f61a')
+    version('1.2.4', sha256='c0485dba3b7c285599809e058c83f33b5efa9522d20d9f980423410604207f61')
 
-    depends_on('python@3:', type=('build', 'run'))
-    depends_on('py-setuptools', type='build')
-    depends_on('py-cython', type='build')
     depends_on('py-wheel@0.28.0:', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-mikado/package.py]:
```suggestion
    depends_on('py-wheel@0.28.0:', type='build')
```

File: var/spack/repos/builtin/packages/py-mikado/package.py [hunk 7]
```diff
@@ -12,34 +12,29 @@ class PyMikado(PythonPackage):
...
-    depends_on('py-pytest', type=('build', 'run'))
+    depends_on('py-nose', type=('build', 'run'))
     depends_on('py-pyfaidx', type=('build', 'run'))
     depends_on('py-scikit-learn@0.17.0:', type=('build', 'run'))
-    depends_on('py-scipy@1.0.0:', type=('build', 'run'))
+    depends_on('py-scipy@0.15.0:', type=('build', 'run'))
+    depends_on('py-python-magic', type=('build', 'run'))
     depends_on('py-drmaa', type=('build', 'run'))
     depends_on('snakemake', type=('build', 'run'))
     depends_on('py-docutils@0.13.2:', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-mikado/package.py]:
```suggestion
    depends_on('py-docutils@:0.13.0,0.13.2:', type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-mikado/package.py [hunk 8]
```diff
@@ -12,34 +12,29 @@ class PyMikado(PythonPackage):
...
     depends_on('py-tabulate', type=('build', 'run'))
-    depends_on('py-msgpack@1.0.0:', type=('build', 'run'))
-    depends_on('py-python-rapidjson@0.8:', type=('build', 'run'))
-    depends_on('py-pandas', type=('build', 'run'))
-    depends_on('py-jsonref', type=('build', 'run'))
-    depends_on('py-pysam@0.15.3:', type=('build', 'run'))
-    depends_on('py-toml', type=('build', 'run'))
-    depends_on('py-tomlkit', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-mikado/package.py:45]: Missing a dep on `py-tabulate`

File: var/spack/repos/builtin/packages/py-mikado/package.py [hunk 9]
```diff
@@ -12,34 +12,29 @@ class PyMikado(PythonPackage):
...
+    depends_on('py-ujson', type=('build', 'run'))
+    depends_on('py-simplejson', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-mikado/package.py:39]: Missing:
```python
depends_on('py-typing', when='^python@:3.4', type=('build', 'run'))
```

**Review by adamjstewart — CHANGES_REQUESTED** [review]

**Comment by t-nojiri** [comment]
I have fixed what was pointed out in the review.

File: var/spack/repos/builtin/packages/py-mikado/package.py (hunk 8 above)
Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-mikado/package.py:45]: Still missing

Comment (t-nojiri) [reply var/spack/repos/builtin/packages/py-mikado/package.py:45]: I added a py-tabulate depend.

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**